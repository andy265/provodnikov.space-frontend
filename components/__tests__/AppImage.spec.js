const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
const AppImage = require('~/components/AppImage')

describe('AppImage.vue', () => {
  let wrapper = shallow(AppImage, {
    propsData: {
      src: 'http://localhost:3000/some_img.jpg',
      alt: 'description'
    }
  })

  it('has same HTML structure', () => {
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('emit click event', () => {
    wrapper.find('img').trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
  })

  it('emit error event', () => {
    wrapper.find('img').trigger('error')
    expect(wrapper.emitted().error).toBeTruthy()
  })
})
