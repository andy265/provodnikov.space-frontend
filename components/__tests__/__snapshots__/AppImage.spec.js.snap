// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppImage.vue has same HTML structure 1`] = `
<div class="app-image">
    <div class="app-image__inner">
        <img src="http://localhost:3000/some_img.jpg" alt="description" class="app-image__image">
    </div>
</div>
`;
